integration/graphql-type-factories.ts

```diff
@@ -38,6 +38,7 @@ export interface AuthorOptions {
 export function newAuthor(options: AuthorOptions = {}, cache: Record<string, any> = {}): Author {
   const o = (options.__typename ? options : cache["Author"] = {}) as Author;
   (cache.all ??= new Set()).add(o);
+  cache["Author"] ??= o;
```

Comment:
Are these necessary because of the "pass __typename in the POJO param of a new interface"?

I.e. previously I think we assumed __typename would only exist on factory-created instances, so they would be "already cached"...

But wtf was this tertiary doing...always setting cache[Author] to the /last/ created author? That seems backwards...

Reply:
actually don't think these are necessary for the fix